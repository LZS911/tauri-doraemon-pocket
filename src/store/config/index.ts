import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import CONSTANT from '../../common/constant';
import {
  ThemeModeEnum,
  ColorSchemeEnum,
  FontFamilyEnum,
} from '../../common/enum';
import { SupportLanguage } from '../../locale';
export interface IUserConfigState {
  currentThemeMode: ThemeModeEnum;
  currentColorScheme: ColorSchemeEnum;
  lang: SupportLanguage;
  fontFamily: FontFamilyEnum;
}

export const userConfigInitialState: IUserConfigState = {
  currentThemeMode: ThemeModeEnum.Light,
  currentColorScheme: ColorSchemeEnum.Blue,
  lang: SupportLanguage.zhCN,
  fontFamily: FontFamilyEnum.Inter,
};
const config = createSlice({
  name: 'config',
  initialState: userConfigInitialState,
  reducers: {
    setCurrentThemeMode: (state, action: PayloadAction<ThemeModeEnum>) => {
      state.currentThemeMode = action.payload;
    },
    setCurrentColorScheme: (state, action: PayloadAction<ColorSchemeEnum>) => {
      state.currentColorScheme = action.payload;
    },
    setCurrentLang: (state, action: PayloadAction<SupportLanguage>) => {
      state.lang = action.payload;
    },
    setFontFamily: (state, action: PayloadAction<FontFamilyEnum>) => {
      state.fontFamily = action.payload;
    },
  },
});
export const {
  setCurrentThemeMode,
  setCurrentColorScheme,
  setCurrentLang,
  setFontFamily,
} = config.actions;
export default config.reducer;
