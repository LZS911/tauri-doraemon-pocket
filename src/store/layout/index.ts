import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import CONSTANT from '../../common/constant';
export interface ILayoutState {
  isExpandSider: boolean;
  siderWidth: number;
  isDrawerSider: boolean;
}
export const layoutInitialState: ILayoutState = {
  isExpandSider: false,
  siderWidth: CONSTANT.SIDER_WIDTH,
  isDrawerSider: false,
};
const layout = createSlice({
  name: 'layout',
  initialState: layoutInitialState,
  reducers: {
    setIsExpandSider: (state, action: PayloadAction<boolean>) => {
      state.isExpandSider = action.payload;
    },
    changeSiderWidth: (state, action: PayloadAction<number>) => {
      state.siderWidth = action.payload;
    },
    setIsDrawerSider: (state, action: PayloadAction<boolean>) => {
      state.isDrawerSider = action.payload;
    },
    resetLayoutState: () => layoutInitialState,
  },
});
export const {
  setIsExpandSider,
  changeSiderWidth,
  resetLayoutState,
  setIsDrawerSider,
} = layout.actions;
export default layout.reducer;
