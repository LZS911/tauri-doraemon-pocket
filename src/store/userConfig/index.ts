import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import CONSTANT from '../../common/constant';
import {
  ThemeModeEnum,
  ColorSchemeEnum,
  FontFamilyEnum,
} from '../../common/enum';
import { SupportLanguage } from '../../locale';
import LocalStorageWrapper from '../../utils/LocalStorageWrapper';
export interface IUserConfigState {
  username: string;
  emailAddress: string;
  userProfile: string;
  isLogin: boolean;
  token: string;
  currentThemeMode: ThemeModeEnum;
  currentColorScheme: ColorSchemeEnum;
  lang: SupportLanguage;
  fontFamily: FontFamilyEnum;
}
export type IUserLoginInfo = Pick<
  IUserConfigState,
  'username' | 'emailAddress' | 'userProfile'
>;
export type IUserLoginState = Pick<IUserConfigState, 'token' | 'isLogin'>;
const localToken = LocalStorageWrapper.get(CONSTANT.TOKEN);
export const userConfigInitialState: IUserConfigState = {
  username: '',
  emailAddress: '',
  userProfile: '',
  isLogin: !!localToken,
  token: localToken ?? '',
  currentThemeMode: LocalStorageWrapper.getOrDefault(
    CONSTANT.THEME_MODE,
    ThemeModeEnum.Light
  ),
  currentColorScheme: LocalStorageWrapper.getOrDefault(
    CONSTANT.COLOR_SCHEME,
    ColorSchemeEnum.Blue
  ),
  lang: LocalStorageWrapper.getOrDefault(
    CONSTANT.LANGUAGE_STORAGE_KEY,
    SupportLanguage.zhCN
  ),
  fontFamily: LocalStorageWrapper.getOrDefault(
    CONSTANT.FONT_FAMILY,
    FontFamilyEnum.Inter
  ),
};
const userConfig = createSlice({
  name: 'userConfig',
  initialState: userConfigInitialState,
  reducers: {
    setLoginUserInfo: (state, action: PayloadAction<IUserLoginInfo>) => {
      state.username = action.payload.username;
      state.emailAddress = action.payload.emailAddress;
      state.userProfile = action.payload.userProfile;
    },
    setLoginState: (state, action: PayloadAction<IUserLoginState>) => {
      state.isLogin = action.payload.isLogin;
      state.token = action.payload.token;
      localStorage.setItem(CONSTANT.TOKEN, action.payload.token);
    },
    setCurrentThemeMode: (state, action: PayloadAction<ThemeModeEnum>) => {
      state.currentThemeMode = action.payload;
      LocalStorageWrapper.set(CONSTANT.THEME_MODE, action.payload);
    },
    setCurrentColorScheme: (state, action: PayloadAction<ColorSchemeEnum>) => {
      state.currentColorScheme = action.payload;
      LocalStorageWrapper.set(CONSTANT.COLOR_SCHEME, action.payload);
    },
    setCurrentLang: (state, action: PayloadAction<SupportLanguage>) => {
      state.lang = action.payload;
      LocalStorageWrapper.set(CONSTANT.LANGUAGE_STORAGE_KEY, action.payload);
    },
    setFontFamily: (state, action: PayloadAction<FontFamilyEnum>) => {
      state.fontFamily = action.payload;
      LocalStorageWrapper.set(CONSTANT.FONT_FAMILY, action.payload);
    },
  },
});
export const {
  setLoginUserInfo,
  setLoginState,
  setCurrentThemeMode,
  setCurrentColorScheme,
  setCurrentLang,
  setFontFamily,
} = userConfig.actions;
export default userConfig.reducer;
