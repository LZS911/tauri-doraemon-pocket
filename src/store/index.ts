import { configureStore, Store } from '@reduxjs/toolkit';
import userConfig from './userConfig';
import layout from './layout';
const store: Store = configureStore({
  reducer: {
    userConfig,
    layout,
  },
});
export type IReduxState = ReturnType<typeof store.getState>;
export default store;
