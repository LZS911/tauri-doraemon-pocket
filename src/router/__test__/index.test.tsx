import { act, render } from '@testing-library/react';
import { useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import useResizeObserver from 'use-resize-observer';
import { userConfigInitialState } from '../../store/userConfig';
import RouterComponent from '../index';

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
  };
});

jest.mock('use-resize-observer', () => {
  return {
    default: jest.fn(),
  };
});

const useSelectorMock: jest.Mock = useSelector as jest.Mock;
const useDispatchMock: jest.Mock = useDispatch as jest.Mock;
const useResizeObserverMock: jest.Mock = useResizeObserver as jest.Mock;
const dispatchMock = jest.fn();

beforeEach(() => {
  jest.useFakeTimers();
  useResizeObserverMock.mockImplementation(() => {
    return {
      ref: useRef(),
    };
  });
  useDispatchMock.mockImplementation(() => dispatchMock);
  useSelectorMock.mockImplementation((selector) => {
    return selector({
      userConfig: {
        isLogin: false,
        token: '',
      },
    });
  });
});

afterEach(() => {
  jest.clearAllMocks();
  jest.clearAllTimers();
  jest.useRealTimers();
});

describe('test router', () => {
  test('should match snapshot when login state is invalid', async () => {
    const { container } = render(<RouterComponent />);
    await act(async () => {
      jest.advanceTimersByTime(800);
    });

    expect(dispatchMock).toBeCalledTimes(1);
    expect(dispatchMock).toBeCalledWith({
      //todo
      payload: {
        emailAddress: 'demo@gamil.com',
        userProfile: 'Default user',
        username: 'Gll Ly',
      },
      type: 'userConfig/setLoginUserInfo',
    });
    // expect(container).toMatchSnapshot();
  });

  test('should match snapshot when login state is valid', async () => {
    useSelectorMock.mockImplementation((selector) => {
      return selector({
        userConfig: {
          ...userConfigInitialState,
          isLogin: true,
          token: 'abcd',
        },
        layout: {
          isExpandSider: false,
          siderWidth: 60,
          isDrawerSider: false,
        },
      });
    });
    render(<RouterComponent />);
    await act(async () => {
      jest.advanceTimersByTime(800);
    });
  });
});
