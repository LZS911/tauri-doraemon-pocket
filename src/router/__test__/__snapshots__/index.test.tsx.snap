// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test router should match snapshot for registerRouter 1`] = `
Array [
  <Route
    element={
      <React.Suspense
        fallback={<HeaderProgress />}
      >
        <UNDEFINED />
      </React.Suspense>
    }
    path="/dashboard"
  />,
  <Route
    element={
      <React.Suspense
        fallback={<HeaderProgress />}
      >
        <UNDEFINED />
      </React.Suspense>
    }
    path="/chat"
  />,
  <Route
    element={
      <React.Suspense
        fallback={<HeaderProgress />}
      >
        <UNDEFINED />
      </React.Suspense>
    }
    path="/component/input"
  />,
  <Route
    element={
      <React.Suspense
        fallback={<HeaderProgress />}
      >
        <UNDEFINED />
      </React.Suspense>
    }
    path="/component/table"
  />,
  <Route
    element={
      <React.Suspense
        fallback={<HeaderProgress />}
      >
        <UNDEFINED />
      </React.Suspense>
    }
    path="/404"
  />,
]
`;

exports[`test router should match snapshot for registerRouter 2`] = `
Array [
  <Route
    element={
      <React.Suspense
        fallback={<HeaderProgress />}
      >
        <UNDEFINED />
      </React.Suspense>
    }
    path="/login"
  />,
]
`;
